src/util.hpp

```diff
@@ -62,9 +62,9 @@ namespace rocshmem {
 #endif
 
 #ifdef DEBUG
-#define GPU_DPRINTF(...)      \
-  do {                        \
-    gpu_dprintf(__VA_ARGS__); \
+#define GPU_DPRINTF(...)                                         \
+  do {                                                           \
+    gpu_dprintf("WG (%u, %u, %u) TH (%u, %u, %u) " __VA_ARGS__); \
```

Comment:
Would it be worth add which PE we are printing from? Not sure if this will be useful if we initialize rocSHMEM from a sub-communicator, since there will be multiple PEs with the same rank. So this comment is optional

```suggestion
    gpu_dprintf("[%d] WG (%u, %u, %u) TH (%u, %u, %u) " __VA_ARGS__); \
```

Reply:
I thought about that, we have `mpiexec --output rank` to show the global pe-rank, we could show the caller my-pe in the team here?

Reply:
I forgot about that flag. This probably isn't needed